Sources/Swift/Tools/LoadValidator.swift

```diff
@@ -0,0 +1,90 @@
+@_implementationOnly import _SentryPrivate
+import Darwin
+import Foundation
+import MachO
+
+@objc
+@_spi(Private) public final class LoadValidator: NSObject {
+    // Any class should be fine
+    static let targetClassName = "PrivateSentrySDKOnly"
+    
+    @objc
+    @_spi(Private) public class func validateSDKPresenceIn(_ image: SentryBinaryImageInfo) {
+        internalValidateSDKPresenceIn(image, objcRuntimeWrapper: SentryDefaultObjCRuntimeWrapper.sharedInstance())
+    }
+    
+    /**
+     * This synchronous version is intended to be used in tests.
+     * It uses a higher QoS and wait for the result
+     */
+    @discardableResult class func validateSDKPresenceInSync(_ image: SentryBinaryImageInfo, objcRuntimeWrapper: SentryObjCRuntimeWrapper) -> Bool {
+        var result = false
+        let semaphore = DispatchSemaphore(value: 0)
+        internalValidateSDKPresenceIn(image, objcRuntimeWrapper: objcRuntimeWrapper, qos: .userInitiated) { duplicateFound in
+            result = duplicateFound
+            semaphore.signal()
+        }
+        semaphore.wait()
+        return result
+    }
+    
+    class private func internalValidateSDKPresenceIn(_ image: SentryBinaryImageInfo, objcRuntimeWrapper: SentryObjCRuntimeWrapper, qos: DispatchQoS.QoSClass = .background, resultHandler: ((Bool) -> Void)? = nil) {
+        DispatchQueue.global(qos: qos).async {
+            var duplicateFound = false
+            defer {
+                resultHandler?(duplicateFound)
+            }
+            let systemLibraryPath = "/usr/lib/"
+#if targetEnvironment(simulator)
+            let ignoredPath = "/Library/Developer/CoreSimulator/Profiles/Runtimes/"
+#else
+            let ignoredPath = "/System/Library/"
+#endif
+            let imageName = image.name
+            guard !imageName.contains(ignoredPath) && !imageName.hasPrefix(systemLibraryPath) else {
+                return
+            }
+            
+            let loadValidatorAddress = self.getCurrentFrameworkTextPointer()
+            let imageAddress = image.address
+            let imageSize = image.size
+            let loadValidatorAddressValue = UInt(bitPattern: loadValidatorAddress)
+            let isCurrentImageContainingLoadValidator = (loadValidatorAddressValue >= imageAddress) && (loadValidatorAddressValue < (imageAddress + imageSize))
```

Comment:
`m`: Couldn't we add an early return if this is true? Below we continue when this is true and `name == self.targetClassName`.

Reply:
We could find a different name in the same image

Reply:
It would be great to add this info as a comment.